packages/types/src/payment/processors.ts

```diff
@@ -0,0 +1,310 @@
+import { PaymentSessionDTO, PaymentSessionStatus } from "./common"
+
+/**
+ * @interface
+ *
+ * A payment's context.
+ */
+export type PaymentProcessorContext = {
+  /**
+   * The payment's billing address.
+   */
+  billing_address?: Record<string, unknown> | null // TODO: revisit types
+  /**
+   * The customer's email.
+   */
+  email: string
+  /**
+   * The selected currency code, typically associated with the customer's cart.
+   */
+  currency_code: string
+  /**
+   * The payment's amount.
+   */
+  amount: number
+  /**
+   * The ID of the resource the payment is associated with. For example, the cart's ID.
+   */
+  resource_id: string
+  /**
+   * The customer associated with this payment.
+   */
+  customer?: Record<string, unknown> // TODO: type
+  /**
+   * The cart's context.
+   */
+  context: Record<string, unknown>
+  /**
+   * If the payment session hasn't been created or initiated yet, it'll be an empty object.
+   * If the payment session exists, it'll be the value of the payment session's `data` field.
+   */
+  paymentSessionData: Record<string, unknown>
```

Comment:
todo: Snake case this property or camel all others.
```suggestion
  payment_session_data: Record<string, unknown>
```